src/app/components/chat-component/chat-component.component.ts

```diff
@@ -0,0 +1,89 @@
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {MessageService} from "../../../services/messageservice";
+import {ApiService} from "../../../services/api";
+
+@Component({
+  selector: 'app-chat-component',
+  templateUrl: './chat-component.component.html',
+  styleUrls: ['./chat-component.component.css']
+})
+export class ChatComponentComponent {
+  public urlParams = '';
+  public chat: any = { message: ''};
+  public chatMessages: any[] = [];
+  public author = '';
+  @Input() options: any = { };
+  @Output() doConfirm: any = new EventEmitter();
+
+  constructor(
+    public messageService: MessageService,
+    public api: ApiService
+  ) { }
+
+  ngOnInit(){
+    this.loadChat();
+  }
+
+  /**
+   * load chat history
+   */
+  public loadChat() {
+    const appUser: any = this.messageService.getUserData();
+    console.log('appUser', appUser.fullname);
+    this.urlParams = `?page=1&limit=10&order={"createdAt":-1}`;
+    console.log(this.urlParams);
+    this.messageService.getMessage(this.urlParams).subscribe(
+      (responseLoad: any) => {
+        console.log('responseLoad', responseLoad);
+        for (let i = responseLoad.rows.length - 1; i >= 0; i--) {
```

Comment:
Move this block of functionality to separate function like "public fetchResponseToMessages(responseLoad) {...}"